main.py

```diff
@@ -375,43 +406,61 @@ def update_preferences():
     if cur_user is None:
         return jsonify(respo='user not found')
     cur_report_preferences = db_session.query(ReportPreferences).filter(User.id == cur_id).first()
+    cur_general_preferences = db_session.query(GeneralPreferences).filter(User.id == cur_id).first()
     if request.method == "GET":
-        if cur_report_preferences is None:
-            return jsonify(lat='', lon='', distance='', accident_severity='0', history_report=False)
+        if cur_report_preferences is None and cur_general_preferences is None:
+            return jsonify(accident_severity='0', prefAccidentsLms=True, prefAccidentsIhud=True, produceAccidentsReport=False)
         else:
-            return jsonify(lat=cur_report_preferences.latitude, lon=cur_report_preferences.longitude, distance=cur_report_preferences.radius, \
-                accident_severity=cur_report_preferences.minimum_severity, history_report=cur_report_preferences.historical_report)
+            resource_types = cur_general_preferences.resource_type.split(',')
+            if cur_report_preferences is None:
+                return jsonify(accident_severity=cur_general_preferences.minimum_displayed_severity, prefResourceTypes=resource_types, produceAccidentsReport=False)
+            else:
+                return jsonify(accident_severity=cur_general_preferences.minimum_displayed_severity, prefResourceTypes=resource_types, produceAccidentsReport=True, \
+                               lat=cur_report_preferences.latitude, lon=cur_report_preferences.longitude, prefRadius=cur_report_preferences.radius, \
+                               prefAccidentSeverityForReport=cur_report_preferences.minimum_severity, how_many_months_back=cur_report_preferences.how_many_months_back)
     else:
-        jsonData = request.get_json(force=True)    
-        lat = jsonData['lat']
-        lon = jsonData['lon']
-        distance = jsonData['distance']
-        accident_severity = jsonData['accident_severity']
-        history_report = jsonData['history_report']
-    
+        json_data = request.get_json(force=True)
+        accident_severity = json_data['accident_severity']
+        resources = json_data['prefResourceTypes']
```

Comment:
Sorry for being stubborn, but case convention should be consistent even in json fields. Use underscores rather than camel case.